app/representers/api/image_representer.rb

```diff
@@ -2,12 +2,30 @@
 
 class Api::ImageRepresenter < Api::BaseRepresenter
 
+  def self_url(represented)
+    if represented.is_a?(StoryImage)
+      api_story_story_image_path(represented.story, represented)
+    elsif represented.is_a?(SeriesImage)
+      api_series_series_image_path(represented.series, represented)
+    elsif represented.is_a?(AccountImage)
+      api_account_account_image_path(represented.account, represented)
+    elsif represented.is_a?(UserImage)
+      api_user_user_image_path(represented.user)
+    else
+      super
+    end
+  end
+
   property :id, writeable: false
-  property :filename
+  property :filename, writeable: false
   property :size
   property :caption
   property :credit
 
+  # provide either an accessible url or the file itself for upload
+  property :upload, readable: false
+  property :file, readable: false
```

Comment:
This will change with background processing.

But: `upload` (named to match the existing audio-file `upload` param) is currently aliased to `remote_file_url`.  So you give it an s3 url, it downloads the file, and sets `image.file = downloaded_file`.  When we switch over to background processing, the param name can probably stay the same.  But instead of CMS downloading the file, it'll just wait for the lambda to finish working.

I left the `file` param, in case I wanted to directly POST the file to cms.  Seems like we never will, though.